design-document.md

```diff
@@ -30,6 +30,62 @@ practical.]]
 
 ## System constraints
 
+Prio has inherent constraints derived from the tradeoff between privacy
+guarantees and computational complexity. These tradeoffs influence how
+applications may choose to utilize services implementing the specification.
+
+### Data types have limited resolution due to input-validation complexity
+
+Privacy comes at the cost of computational complexity. While affine-aggregatable
+encodings can compute many useful statistics, they require more bandwidth and
+CPU cycles to account for finite-field arithmetic during input-validation. The
+throughput of the system bounds the volume of data processed, which is
```

Comment:
What is the throughput of the system? How many inputs/second can be processed?

Taking a step back: Should we have a section where we discuss this and other key performance indicators that we're trying to be mindful of? (Time between batch start and batch end, where the aggregate is available, comes to mind, for example.)

Reply:
I think inputs/second is a reasonable definition of throughput, which is dependent on input parameters and the resources allocated for work. I guess "rate of work" would do here too. 

Using a maybe not so great physics analogy to explain in a different frame:
* verifying input is work
* throughput is power (rate of work)
* compute-time is energy (capacity for work)
* the amount of data that can be processed is constrained by throughput and available compute-time

> Should we have a section where we discuss this and other key performance indicators that we're trying to be mindful of?

It's probably a good idea. Maybe something to consider in the requirements or a separate performance section. I imagine an implementation would also want to monitor some of these things (e.g time to completion, error counts).  This section could also be expanded for discussion.

Reply:
> It's probably a good idea. Maybe something to consider in the requirements or a separate performance section. I imagine an implementation would also want to monitor some of these things (e.g time to completion, error counts). This section could also be expanded for discussion.

Both seem fine. Maybe just mark an OPEN ISSUE here for us to resolve later?